ui/test/services/project_service_test.dart

```diff
@@ -36,20 +36,32 @@ void main() {
     });
 
     group('Projects query', () {
-      test('queries list of projects', () async {
-        when(client.getProjects())
-            .thenAnswer((_) => Future.value([Project(id: projectId)]));
+      test('query to list all projects when no filter applied', () async {
+        when(client.findProjectsBySearchFragment(''))
+            .thenAnswer((_) => Future.value([
+                  Project(id: projectId, title: 'ProjectA'),
+                  Project(id: projectId, title: 'ProjectB')
```

Comment:
Limit the response to a single project to limit the cognitive distraction. (The list of projects is a detail for this test.)